### ISSUE TITLE
[BUG] Confluence dependency to mark an incident as resolved

### ISSUE THREAD

**Issue by y0hnn** [echoboomer/incident-bot#519]
**Describe the bug**
When switching the status of an incident to `resolved`, the pod is crashing. The error : 

```
{"name": "slack_bolt.App", "asctime": "2024-05-13 15:39:26,342", "levelname": "ERROR", "filename": "handler.py", "message": "Error: 'NoneType' object has no attribute 'get'", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.11/site-packages/slack_bolt/listener/thread_runner.py\", line 120, in run_ack_function_asynchronously\n    listener.run_ack_function(request=request, response=response)\n  File \"/usr/local/lib/python3.11/site-packages/slack_bolt/listener/custom_listener.py\", line 50, in run_ack_function\n    return self.ack_function(\n           ^^^^^^^^^^^^^^^^^^\n  File \"/incident-bot/bot/slack/handler.py\", line 437, in handle_incident_set_status\n    asyncio.run(inc_actions.set_status(action_parameters=parse_action(body)))\n  File \"/usr/local/lib/python3.11/asyncio/runners.py\", line 190, in run\n    return runner.run(main)\n           ^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/asyncio/runners.py\", line 118, in run\n    return self._loop.run_until_complete(task)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/asyncio/base_events.py\", line 654, in run_until_complete\n    return future.result()\n           ^^^^^^^^^^^^^^^\n  File \"/incident-bot/bot/incident/actions.py\", line 415, in set_status\n    .get(\"auto_create_postmortem\")\n     ^^^\nAttributeError: 'NoneType' object has no attribute 'get'"}
```

**Version**
1.10.3

**To Reproduce**
1. Disable Confluence integration
2. Create an incident
3. Set it as resolved

It seems to be coming from this line https://github.com/echoboomer/incident-bot/blob/main/backend/bot/incident/actions.py#L414

**Comment by echoboomer** [comment]
@y0hnn can you please share a sanitized version of your configuration?

**Comment by y0hnn** [comment]
yes : 

```
configMap:
  create: true
  data:
    platform: slack
    digest_channel: incidents
    roles:
      incident_commander: "The Incident Commander is the decision maker during a major incident, delegating tasks and listening to input from subject matter experts in order to bring the incident to resolution. They become the highest ranking individual on any major incident call, regardless of their day-to-day rank. Their decisions made as commander are final.\\n\\nYour job as an Incident Commander is to listen to the call and to watch the incident Slack room in order to provide clear coordination, recruiting others to gather context and details. You should not be performing any actions or remediations, checking graphs, or investigating logs. Those tasks should be delegated.\\n\\nAn IC should also be considering next steps and backup plans at every opportunity, in an effort to avoid getting stuck without any clear options to proceed and to keep things moving towards resolution."
      communications_liaison: "The purpose of the Communications Liaison is to be the primary individual in charge of notifying our customers of the current conditions, and informing the Incident Commander of any relevant feedback from customers as the incident progresses.\\n\\nIt's important for the rest of the command staff to be able to focus on the problem at hand, rather than worrying about crafting messages to customers.\\n\\nYour job as Communications Liaison is to listen to the call, watch the incident Slack room, and track incoming customer support requests, keeping track of what's going on and how far the incident is progressing (still investigating vs close to resolution).\\n\\nThe Incident Commander will instruct you to notify customers of the incident and keep them updated at various points throughout the call. You will be required to craft the message, gain approval from the IC, and then disseminate that message to customers."
    severities:
      P1: 'This signifies a critical production scenario that impacts most or all users with a major impact on SLAs. This is an all-hands-on-deck scenario that requires swift action to restore operation. Customers must be notified.'
      P2: 'This signifies a significant production degradation scenario impacting a large portion of users.'
      P3: 'This signifies a minor production scenario that may or may not result in degradation. This situation is worth coordination to resolve quickly but does not indicate a critical loss of service for users.'
      P4: 'This signifies an ongoing investigation. This incident has not been promoted to P3 yet, indicating there may be little to no impact, but the situation warrants a closer look. This is diagnostic in nature. This is the default setting for a new incident.'
    incident_reminders:
      qualifying_severities:
        - P1
      rate: 30
    statuses:
      - investigating
      - identified
      - monitoring
      - resolved
    options:
      channel_naming:
        time_format_in_channel_name: '%Y-%m-%d'
      channel_topic:
        default: 'This is the default incident channel topic. You can edit it in settings.'
      conference_bridge_link: "https://discord.com/"
      timezone: UTC
    integrations:
      atlassian:
        jira:
          project: INC
          issue_types: ['Incident']
          priorities: ['++++', '+++', '++', '+']
          auto_create_incident: true
          auto_create_incident_type: Incident
          labels:
            - incident
    links: 
      - title: "Incidents list"
        url: https://www.notion.so/
envFromSecret:
  enabled: true
  secretName: incident-bot-secret
ingress:
  enabled: false
  annotations:
    kubernetes.io/ingress.class: nginx
    cert-manager.io/cluster-issuer: letsencrypt-prod
  hosts:
    - host: incident-bot.myhost.com
      paths:
        - path: /
          pathType: ImplementationSpecific
podDisruptionBudget:
  enabled: true
  minAvailable: 1
database:
  enabled: true
  password: mypassword
```

**Comment by echoboomer** [comment]
@y0hnn would you mind bumping to `v1.10.7` and seeing if the error is resolved?

**Comment by y0hnn** [comment]
I confirm it's now working, thanks!

**Comment by echoboomer** [comment]
@y0hnn I just released `v1.12.0` which should resolve this. Please let me know if not. Thanks!
